Lib/Database/Database.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * Database class file.
+ *
+ * @package Freyja\Database
+ * @copyright 2016 SqueezyWeb
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ */
+
+namespace Freyja\Database;
+use Freyja\Database\Driver;
+use Freyja\Database\Query;
+use Freyja\Exceptions\InvalidArgumentException as InvArgExcp;
+use RuntimeException;
+
+/**
+ * Database class.
+ *
+ * @package Freyja\Database
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ * @version 1.0.0
+ */
+class Database {
+  /**
+   * Driver used to execute queries.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var Driver
+   */
+  private $driver;
+
+  /**
+   * Global instance.
+   *
+   * @since 1.0.0
+   * @access private
+   * @static
+   * @var Database
+   */
+  private static $global_instance;
+
+  /**
+   * Last executed query.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var Query
+   */
+  private $last;
+
+  /**
+   * Class constructor.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param Driver $driver Driver used to execute queries.
+   */
+  public function __construct(Driver $driver) {
+    $this->driver = $driver;
+  }
+
+  /**
+   * Connect to database.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $host Database address.
+   * @param string $database Database name.
+   * @param string $username Access username.
+   * @param string $password Access password.
+   * @return self
+   *
+   * @throws Freyja\Exceptions\InvalidArgumentException if one of the arguments
+   * isn't a string.
+   */
+  public function connect($host, $database, $username, $password) {
+    try {
+      $this->driver->connect($host, $database, $username, $password);
+    } catch (Exception $e) {
+      throw $e;
+    }
+
+    return $this;
+  }
+
+  /**
+   * Set global instance.
+   *
+   * Set this instance as global.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @return self
+   */
+  public function setGlobal() {
+    self::$global_dinstance = $this;
+    return $this;
+  }
+
+  /**
+   * Retrieve global instance.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @return Database|boolean Global instance. False if global instance isn't
+   * set.
+   */
+  public function getGlobal() {
+    if (!isset(self::$global_instance))
```

Comment:
This check is wrong, since `self::$global_instance` is **always** set.
Anyway, you can simply return `null` if the global instance has not been initialized, so in this method you simply do a:
```php
return self::$global_instance
```